### ISSUE TITLE
Logcat Spamming "Failed to open APK '/system/app/NavigationBarModeGestura/NavigationBarModeGesturalOverlay.apk': I/O error"  in LineageOS 20 & 21 for LG V20

### ISSUE THREAD

**Issue by blurpng** [Magisk-Modules-Alt-Repo/HideNavBar#63]
not sure if this is intended behavior, as with my other phone the logcat is clean without spamming this error while this specific phone with this rom does it.

It seems to trigger every time an application is opened/used, wonder if this might cause issues down the line

[logcata14gestures.txt](https://github.com/Magisk-Modules-Alt-Repo/HideNavBar/files/14318789/logcata14gestures.txt)
[logcatpostinstall.txt](https://github.com/Magisk-Modules-Alt-Repo/HideNavBar/files/14318790/logcatpostinstall.txt)

**Comment by DanGLES3** [comment]
> not sure if this is intended behavior, as with my other phone the logcat is clean without spamming this error while this specific phone with this rom does it.
> 
> It seems to trigger every time an application is opened/used, wonder if this might cause issues down the line
> 
> [logcata14gestures.txt](https://github.com/Magisk-Modules-Alt-Repo/HideNavBar/files/14318789/logcata14gestures.txt) [logcatpostinstall.txt](https://github.com/Magisk-Modules-Alt-Repo/HideNavBar/files/14318790/logcatpostinstall.txt)

Are you using KernelSU? Or Magisk Kitsune with the Sulist function enabled?

**Comment by blurpng** [comment]
> > not sure if this is intended behavior, as with my other phone the logcat is clean without spamming this error while this specific phone with this rom does it.
> > It seems to trigger every time an application is opened/used, wonder if this might cause issues down the line
> > [logcata14gestures.txt](https://github.com/Magisk-Modules-Alt-Repo/HideNavBar/files/14318789/logcata14gestures.txt) [logcatpostinstall.txt](https://github.com/Magisk-Modules-Alt-Repo/HideNavBar/files/14318790/logcatpostinstall.txt)
> 
> Are you using KernelSU or Magisk Kitsune with the Sulist function enabled?
> 
> If it's either of there I suspect it's merely a side effect of the overlay being unmounted/inaccesible to apps without root access
> 
> In short, overlays applied to the system framework tend to propagate to all running apps, but if root is being hidden from the app that means the overlay APK is as well which trigger this message

Nope, i'm running bog standard Magisk 27.0 with zygist & systemless host on this phone
Issue happen in both fullscreen and immersive install modes

The interesting bug i encounter in LOS20 is that it crashes the phone after a while on some apps that often triggers the error, while on LOS21 it just make the same apps refuse to open/stuck in loading (like gcam)

**Comment by DanGLES3** [comment]
> > > not sure if this is intended behavior, as with my other phone the logcat is clean without spamming this error while this specific phone with this rom does it.
> > > It seems to trigger every time an application is opened/used, wonder if this might cause issues down the line
> > > [logcata14gestures.txt](https://github.com/Magisk-Modules-Alt-Repo/HideNavBar/files/14318789/logcata14gestures.txt) [logcatpostinstall.txt](https://github.com/Magisk-Modules-Alt-Repo/HideNavBar/files/14318790/logcatpostinstall.txt)
> > 
> > 
> > Are you using KernelSU or Magisk Kitsune with the Sulist function enabled?
> > If it's either of there I suspect it's merely a side effect of the overlay being unmounted/inaccesible to apps without root access
> > In short, overlays applied to the system framework tend to propagate to all running apps, but if root is being hidden from the app that means the overlay APK is as well which trigger this message
> 
> Nope, i'm running bog standard Magisk 27.0 with zygist & systemless host on this phone Issue happen in both fullscreen and immersive install modes
> 
> The interesting bug i encounter in LOS20 is that it crashes the phone after a while on some apps that often triggers the error, while on LOS21 it just make the same apps refuse to open/stuck in loading (like gcam)

Are these apps on the dentist? Are you using any additional module to hide root?

**Comment by blurpng** [comment]
there are some of them in the denylist, but i think its not related as chrome isnt one of them yet that also triggers it. 
for root hiding i only have shamiko 1.0 and play integrity fix

ill give the other suggestion a shot later on today

**Comment by DanGLES3** [comment]
> there are some of them in the denylist, but i think its not related as chrome isnt one of them yet that also triggers it. for root hiding i only have shamiko 1.0 and play integrity fix
> 
> ill give the other suggestion a shot later on today

Check if the error messages occur with Shamiko disabled as well

**Comment by DanGLES3** [comment]
> so i tried both disabling shamiko solved half the issue but still spews error, removing the idmap fixed the error spam, however re-enabling shamiko brought the errors back so i guess this is the problematic module.
> 
> the curiosity remains though as on my other phone, an lg wing with stock rom, seems to live fine along with shamiko, got any idea what this could be? attached logcats on both removing the idmap & disabling shamiko, removing idmap but keeping shamiko of the V20:
> 
> maybe you know alternate module that wont interfere that acts similarly?
> 
> [los21-27-02-gestures-datares-postremove-wshamiko.txt](https://github.com/Magisk-Modules-Alt-Repo/HideNavBar/files/14408329/los21-27-02-gestures-datares-postremove-wshamiko.txt) [los21-27-02-gestures-datares-postremove.txt](https://github.com/Magisk-Modules-Alt-Repo/HideNavBar/files/14408335/los21-27-02-gestures-datares-postremove.txt)

Is your Shamiko working in whitelist mode?

**Comment by blurpng** [comment]
shamiko is in blacklist mode in the v20

**Comment by DanGLES3** [comment]
> shamiko is in blacklist mode in the v20

Could you try reinstalling the module? As for alternative modules, i'm only aware of Iconify and PIxel Xpert

**Comment by blurpng** [comment]
well that is unfortunate then, since i did the reinstall and it seems to spam the same error. thanks for the help but it seems like the module just doesnt work well on this phone & rom combo

**Comment by DanGLES3** [comment]
> well that is unfortunate then, since i did the reinstall and it seems to spam the same error. thanks for the help but it seems like the module just doesnt work well on this phone & rom combo

Tbh seems more like a Shamiko issue, it seems to be messing with framework overlays somehow, tho you're the first person that reported this issue which is odd

Is your Shamiko uptodate?

**Comment by blurpng** [comment]
it does seem like a shamiko issue mixed with the rom itself, but i cant help it since some apps breaks without it, unless i want to try other magisk forks which hides itself better. 

The difference on the V20 is im using the latest shamiko, 1.0.1, while the wing uses 0.7.5

**Comment by DanGLES3** [comment]
> it does seem like a shamiko issue mixed with the rom itself, but i cant help it since some apps breaks without it, unless i want to try other magisk forks which hides itself better.
> 
> The difference on the V20 is im using the latest shamiko, 1.0.1, while the wing uses 0.7.5

Could be that, unfortunately there isn't a whole lot i can on my end

**Comment by UltraHQ** [comment]
Hi, I'm on kernelsu and experience the same issue. In the logcat, this error is getting spammed a lot.
I don't use shamiko or anything else, just normal ksu.

How do I fix it?
